torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -321,6 +334,13 @@ TORCH_LIBRARY(_c10d_functional, m) {
           c10::DispatchKey::CompositeExplicitAutograd, ::all_reduce_coalesced_),
       {at::Tag::pt2_compliant_tag});
 
+  m.def(
+      "all_gather_into_tensor_out(Tensor input, int group_size, str group_name, *, Tensor(a!) out) -> Tensor(a!)",
```

Comment:
Nice! We can establish the convention that:
- Collectives that modify the input are postfixed with `_`
- Out-variant of collectives are postfixed with `_out`, and the `out` argument is keyword-only